### PR TITLE
D8CORE-1409: Remove duplicate source tag from embedded media.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Text editor filter to fix a core issue. 
- Responsive images in the text editor are outputting `<source ..></source>` which is invalid html as the libxml2 library does not support new html5 tags. 
- see: https://www.drupal.org/project/drupal/issues/1333730
- As of 01/31/2020 there has been no movement in the library to support html5
- This PR adds a filter that we can add to the text formats to do the manual string manipulation on the generated output to correct that.

# Review By (Date)
- End of the month (feb)

# Urgency
- Low, as stuff still works but validators don't like it.

# Steps to Test

1. Check out this branch in our local build of `stanford_profile` 
2. Navigate to and enable the new option in the HTML format `/admin/config/content/formats/manage/stanford_html`
3. Create a new basic page with a WYSIWYG paragraph and insert an image with default settings
4. Save page and view source (not in inspector as that will lie to you.)
5. Validate the `<source>` tag is self-closing.

# Affected Projects or Products
- D8Core and anything that uses Media

# Associated Issues and/or People
- D8CORE-1409
- https://github.com/SU-SWS/acsf-cardinald7/tree/1.x/docroot/profiles/stanford
- @boznik 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: tests/src/Kernel/Plugin/Filter/MediaEmbedMarkupFilterTest.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        ],
+      ])
+      ->addImageStyleMapping('responsive_image_test_module.wide', '1x', [
+        'image_mapping_type' => 'image_style',
+        'image_mapping' => '',
+      ])
+      ->save();
+
+    EntityViewMode::create([
+      'id' => 'media.dupededupe',
```
Comment (pookmish) [inline tests/src/Kernel/Plugin/Filter/MediaEmbedMarkupFilterTest.php]: :rofl:

File: tests/src/Kernel/Plugin/Filter/MediaEmbedMarkupFilterTest.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    'breakpoint',
+    'responsive_image',
+    'responsive_image_test_module',
+    'stanford_media',
+  ];
+
+  /**
+   * @var \Drupal\responsive_image\Entity\ResponsiveImageStyle
+   */
+  protected $RImgStyle;
```
Comment (sherakama) [inline tests/src/Kernel/Plugin/Filter/MediaEmbedMarkupFilterTest.php]: oh I'm not using this and it can be deleted.

**Comment by pookmish** [comment]
Why is there a need to support libxml2?

File: src/Plugin/Filter/MediaEmbedMarkupFilter.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Drupal\stanford_media\Plugin\Filter;
+
+use Drupal\filter\Plugin\FilterBase;
+use Drupal\filter\FilterProcessResult;
+
+/**
+ * @Filter(
```
Comment (pookmish) [inline src/Plugin/Filter/MediaEmbedMarkupFilter.php]: Docblock comment missing

**Comment by sherakama** [comment]
> Why is there a need to support libxml2?

Media embed uses DOM which uses libxml2 to parse and manipulate the wysiwyg text so that the media token is replaced with the rendered output. 

1. DOM parse wysiwyg string
2. Manipulate DOM XML with tools to get rendered token in to place.
3. Reverse the DOM XML back in to a string (this is where the error happens.)
4. Send along the result string to other parsers.

**Comment by sherakama** [comment]
```
is_file(): Unable to find the wrapper "vfs" - did you forget to enable it when you configured PHP?
```

https://www.drupal.org/project/drupal/issues/3072702

**Comment by sherakama** [comment]
@pookmish Danke.

**Comment by imonroe** [comment]
Looks reasonable to me, once coverage test passes.

**Pull request merged**